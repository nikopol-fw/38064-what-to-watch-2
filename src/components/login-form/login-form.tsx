import * as React from 'react';
import {FormEvent} from "react";
import {connect} from 'react-redux';
import {withRouter, RouteComponentProps} from 'react-router-dom';

import {FormLogin} from '../../models/FormLogin';
import {Operation} from '../../reducer/data/data';


interface Props extends RouteComponentProps<any> {
  onAuthorize: (formData: FormLogin) => Promise<any>;
}

export class LoginForm extends React.PureComponent<Props, null> {

  constructor(props: Props) {
    super(props);

    this.handleFormSubmit = this.handleFormSubmit.bind(this);
  }

  private handleFormSubmit(evt: FormEvent<HTMLFormElement>) {
    evt.preventDefault();

    const form = evt.target;

    const formData: FormLogin = {
      email: form[`email`].value,
      password: form[`password`].value,
    };

    this.props.onAuthorize(formData);
  }

  render() {
    return (
      <form className="sign-in__form" onSubmit={this.handleFormSubmit}>
        <div className="sign-in__fields">
          <div className="sign-in__field">
            <input className="sign-in__input" type="email" placeholder="Email address" name="email" id="user-email"
              required
            />
            <label className="sign-in__label visually-hidden" htmlFor="user-email">Email address</label>
          </div>
          <div className="sign-in__field">
            <input className="sign-in__input" type="password" placeholder="Password" name="password" id="user-password"
              required
            />
            <label className="sign-in__label visually-hidden" htmlFor="user-password">Password</label>
          </div>
        </div>
        <div className="sign-in__submit">
          <button className="sign-in__btn" type="submit">Sign in</button>
        </div>
      </form>
    );
  }
}


const mapDispatchToProps = (dispatch) => ({
  onAuthorize: (formData: FormLogin) => dispatch(Operation.authorize(formData)),
});


export default connect(null, mapDispatchToProps)(withRouter(LoginForm));
