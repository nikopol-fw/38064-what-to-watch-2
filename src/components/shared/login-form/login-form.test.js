import React from 'react';
import renderer from 'react-test-renderer';

import {LoginForm} from "./login-form";


const mock = {
  authorize: () => void (0),
};

it(`LoginForm correctly renders afters relaunch`, () => {
  const tree = renderer.create(
      <LoginForm
        authorize={mock.authorize}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
