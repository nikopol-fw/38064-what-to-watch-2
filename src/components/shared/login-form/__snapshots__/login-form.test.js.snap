// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm correctly renders afters relaunch 1`] = `
<form
  className="sign-in__form"
  onSubmit={[Function]}
>
  <div
    className="sign-in__fields"
  >
    <div
      className="sign-in__field"
    >
      <input
        className="sign-in__input"
        id="user-email"
        name="email"
        placeholder="Email address"
        required={true}
        type="email"
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-email"
      >
        Email address
      </label>
    </div>
    <div
      className="sign-in__field"
    >
      <input
        className="sign-in__input"
        id="user-password"
        name="password"
        placeholder="Password"
        required={true}
        type="password"
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-password"
      >
        Password
      </label>
    </div>
  </div>
  <div
    className="sign-in__submit"
  >
    <button
      className="sign-in__btn"
      type="submit"
    >
      Sign in
    </button>
  </div>
</form>
`;
